src/dialer/queueManager.js

```diff
@@ -11,9 +15,53 @@ class DialQueueManager {
    */
   constructor (_switch) {
     this._queue = new Set()
+    this._coldCallQueue = new Set()
     this._dialingQueues = new Set()
     this._queues = {}
     this.switch = _switch
+    this._cleanInterval = retimer(this._clean.bind(this), QUARTER_HOUR)
+  }
+
+  /**
+   * Runs through all queues, aborts and removes them if they
+   * are no longer valid. A queue that is blacklisted indefinitely,
+   * is considered no longer valid.
+   * @private
+   */
+  _clean () {
+    const queues = Object.values(this._queues)
+    queues.forEach(dialQueue => {
+      // Clear if the queue has reached max blacklist
+      if (dialQueue.blackListed === Infinity) {
+        dialQueue.abort()
+        delete this._queues[dialQueue.id]
```

Comment:
What do you think about doing `this._queues[dialQueue.id] = undefined`?

This is more efficient than deleting, but than we will need to verify that keys have a value when iterating the array, so I do not have a strong opinion here.

Reply:
Since this is only running on a 15min interval I think the performance hit should be fairly insignificant and would avoid us needing to check values.